var/spack/repos/builtin/packages/amdfftw/package.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from spack import *
+from spack.pkg.builtin.fftw import FftwBase
+
+
+class Amdfftw(FftwBase):
+    """FFTW (AMD Optimized version) is a comprehensive collection of
+    fast C routines for computing the Discrete Fourier Transform (DFT)
+    and various special cases thereof.
+
+    It is an open-source implementation of the Fast Fourier transform
+    algorithm. It can compute transforms of real and complex-values
+    arrays of arbitrary size and dimension.
+    AMD Optimized FFTW is the optimized FFTW implementation targeted
+    for AMD CPUs.
+    """
+
+    _name = 'amdfftw'
+    homepage = "https://developer.amd.com/amd-aocl/fftw/"
+    url = "https://github.com/amd/amd-fftw/archive/2.2.tar.gz"
+    git = "https://github.com/amd/amd-fftw.git"
+
+    maintainers = ['amd-toolchain-support']
+
+    version('2.2', sha256='de9d777236fb290c335860b458131678f75aa0799c641490c644c843f0e246f8')
+
+    variant(
+        'precision', values=any_combination_of(
+            'float', 'single', 'double', 'long_double',
+            'quad').prohibit_empty_set().with_default('float,double'),
+        description='Build the selected floating-point precision libraries'
+    )
```

Comment:
Is this value the same as in the base class? If so it shouldn't be needed to repeat it.

Reply:
We have additional value 'single' than base fftw. So variant section required.

Reply:
Looking at the recipe it seems `single` is the same as `float`. Did I miss something?

Reply:
Yes, single and float are same. We are trying to provide more flexibility during usage.

Reply:
If they are the same thing, please remove this variant entirely. Having `single` doing the same as `float` will change the hash of the spec (which may result in duplicated builds) but won't do anything to the underlying artifacts.

Reply:
Removed 'single' variant from amdfftw recipe